x-pack/legacy/plugins/code/public/components/integrations/index.tsx

```diff
@@ -8,61 +8,88 @@ import React from 'react';
 import { EuiFlexGroup, EuiPanel, EuiText } from '@elastic/eui';
 
 import { CodeBlock } from '../code_block';
-import { history } from '../../utils/url';
 import { FrameHeader } from './frame_header';
 import { RepoTitle } from './repo_title';
 import { CodeIntegrator } from './code_integrator';
-import { externalFileURI } from './helpers';
 import { frames, Frame, repos, results } from './data';
+import { CodeFlyout } from './code_flyout';
 
 const associateToService = (frame: Frame) => (repo: string) =>
   alert(`repo ${repo} associated with service ${JSON.stringify(frame)}`);
 
 const handleImport = (repo: string) => alert(`import done: ${repo}`);
 
-export const Integrations = () => (
-  <div className="codeIntegrations__container">
-    {frames.map(frame => {
-      const { fileName, lineNumber } = frame;
-      const key = `${fileName}#L${lineNumber}`;
-      const snippet = results[key];
+export const Integrations = () => {
+  const [isFlyoutVisible, setVisible] = React.useState(false);
+  const [flyoutFile, setflyoutFile] = React.useState({
+    repo: 'github.com/Microsoft/TypeScript-Node-Starter',
+    file: 'src/app.ts',
+    revision: 'master',
+  });
 
-      if (snippet) {
-        const { compositeContent, filePath, language, uri } = snippet;
-        const { content, lineMapping } = compositeContent;
-        const fileUrl = externalFileURI(uri, filePath);
+  const closeFlyout = () => {
+    setVisible(false);
+  };
+
+  const showFlyout = (repoUri: string, filePath: string, rev: string = 'master') => {
+    setflyoutFile({
+      repo: 'github.com/Microsoft/TypeScript-Node-Starter',
```

Comment:
We should remove these hardcoded values.

Reply:
Removed. 
Anyone want to see real data in this demo, please import repo `github.com/rylnd/ringside`

Reply:
I shouldn't have based my mock data on a different repo, that's my mistake. I've got a task to rework that data in terms of the Typescript-Node-Starter repo.